### PR TITLE
Feature/commanderAI WIP PR

### PR DESCRIPTION
**Pull request by PiZZAD0X**
**When merged this pull request will:**
- move tasks into config format for task creation, inheritance, multiple entries per task for data about function, responses, characteristics of task
- building garrison changes
- grenade throw function
- nearby enemies change
- verbose debug in core
- exclude target AI from core functions
- target helper functions
- ambient AI fire mode
- create subgroup
- vehicle dismount/support fires

**TODO:**
- commander responses config
- more info used from task config
- work on task functions that are bugged in latest modpack/game updates
- add functions/tasks
- radio responses for enemy -> sent to commander response function
-  vehicle cargo/group inside vehicle not in command
- subgroup usage for assault functions/cargo

**Future Aim:**
- map commander functionality
- remote functions for communicating/commanding with HC AI
- incorporate limited asset usage
- take feedback for purpose of replacing Zeus capabilities whilst having a more mission maker limited game master

### CODE REVIEW

**Comment by joewhite94** [comment]
> subgroup usage for assault functions/cargo

I have something for the assault part, will open a PR into feature/commanderAI in a moment

**Comment by EM-Creations** [comment]
Why does this have a load of changes relating to AO limit etc too?

File: modules/headless_ai/functions/CfgFunctions.hpp [hunk 1]
```diff
@@ -26,14 +26,18 @@ class COMPONENT {
 	class Combat {
 		file = "modules\headless_ai\functions\Combat";
 		class ArmEmptyStatic {};
+		class combatAmbientFire {};
```
Comment (EM-Creations) [inline modules/headless_ai/functions/CfgFunctions.hpp:29]: Style issue, mismatch of camelCase and CapitalCase. Meh.

File: modules/headless_ai/functions/CfgFunctions.hpp [hunk 2]
```diff
@@ -60,6 +63,8 @@ class COMPONENT {
 		class SuppressingShots {};
 		class ThrowGrenade {};
 		class WatchEnemy {};
+		class PrepToFire {};
+		class exitFiring {};
```
Comment (EM-Creations) [inline modules/headless_ai/functions/CfgFunctions.hpp:67]: Style differences for naming.

**Review by joewhite94 — APPROVED** [review]

**Pull request merged**